angular/projects/spark-angular/src/lib/components/sprk-masthead/sprk-masthead.component.ts

```diff
@@ -588,89 +281,69 @@ export class SprkMastheadComponent implements AfterContentInit {
       });
     }
 
-    if (this.isNarrowNavOpen) {
-      classArray.push('sprk-c-Masthead--open');
-    }
-
-    if (this.isScrolled) {
-      classArray.push('sprk-c-Masthead--scroll');
-    }
-
-    if (this.isHidden) {
-      classArray.push('sprk-c-Masthead--hidden');
+    if (
+      this.collapsibleNavDirective &&
+      !this.collapsibleNavDirective.isCollapsed &&
+      this.isNarrowViewport
+    ) {
+      classArray.push('sprk-c-Masthead--is-open');
     }
 
-    return classArray.join(' ');
-  }
-
-  /**
-   * @ignore
-   */
-  getNarrowNavClasses(): string {
-    const classArray: string[] = [];
-
-    if (this.additionalNarrowNavClasses) {
-      this.additionalNarrowNavClasses.split(' ').forEach((className) => {
-        classArray.push(className);
-      });
+    if (this.isPageScrolled) {
+      classArray.push('sprk-c-Masthead--is-scrolled');
     }
 
-    return classArray.join(' ');
-  }
-
-  /**
-   * @ignore
-   */
-  getSecondaryNavClasses(): string {
-    const classArray: string[] = [
-      'sprk-c-Masthead__big-nav-items',
-      'sprk-o-Stack',
-      'sprk-o-Stack--misc-a',
-      'sprk-o-Stack--center-row',
-      'sprk-o-Stack--split@xxs',
-      'sprk-b-List',
-      'sprk-b-List--bare',
-    ];
-
-    if (this.additionalBigNavClasses) {
-      this.additionalBigNavClasses.split(' ').forEach((className) => {
-        classArray.push(className);
-      });
+    if (this.isMastheadHidden && this.isNarrowViewport) {
+      classArray.push('sprk-c-Masthead--is-hidden');
     }
 
     return classArray.join(' ');
   }
 
   /**
-   * @ignore
+   * When the button for the collapsible nav
+   * is clicked this will check if the collapsible
+   * nav is open. If it is open, then it will close it. If it
+   * is closed then it will open it. We run this method
+   * whenever the collapsible nav emits the `isButtonOpen` event.
    */
-  toggleNarrowNav(event): void {
-    event.preventDefault();
-    if (this.isNarrowNavOpen) {
-      this.closeNarrowNav();
+  toggleCollapsibleNav(isButtonOpen: boolean): void {
+    if (!this.collapsibleNavDirective.isCollapsed) {
+      this.closeCollapsibleNav();
     } else {
-      this.openNarrowNav();
+      // The button is now open and we want to open the nav too
+      this.openCollapsibleNav();
     }
   }
 
   /**
-   * @ignore
+   * Adds the correct styles to the body and HTML elements
+   * in order for the collapsible nav to be open.
+   * Sets the `isCollapsed` input on the collapsible nav
+   * to be false so that the directive adds the CSS styles
+   * to show the nav.
    */
-  openNarrowNav(): void {
+  openCollapsibleNav(): void {
     this.renderer.addClass(document.body, 'sprk-u-Overflow--hidden');
     this.renderer.addClass(
       document.body.parentElement,
       'sprk-u-Overflow--hidden',
     );
     this.renderer.addClass(document.body, 'sprk-u-Height--100');
     this.renderer.addClass(document.body.parentElement, 'sprk-u-Height--100');
```

Comment:
Is it safe to use the `!important` utilities here?

Reply:
Good question. I went back and forth on that one too. I felt that it was safe because if for some reason those classes didn't apply the Masthead would break their site.